sdk/cxx/src/address_mapper.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ Copyright 2017 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------
+*/
+#include "sawtooth/address_mapper.h"
+
+#include <exception>
+#include <iostream>
+#include <string>
+
+#include "cryptopp/sha.h"
+#include "cryptopp/filters.h"
+#include "cryptopp/hex.h"
+#include "log4cxx/logger.h"
+
+#include "proto/state_context.pb.h"
+
+namespace sawtooth {
+
+static log4cxx::LoggerPtr  logger(log4cxx::Logger::getLogger
+    ("sawtooth.address_mapper"));
```

Comment:
Is there a way to generate this string instead of hard-coding it so it doesn't need to be updated when the filename changes?